src/serverMain/resources/userFrontend/userFrontend.js

```diff
@@ -46,39 +73,67 @@ function onLoad() {
       }
 
       // Reset result visibility (hide)
-      resultOk.className = resultOk.className.replace("hidden", "") + " hidden";
-      resultFail.className = resultFail.className.replace("hidden", "") + " hidden";
+      resultOkWrapper.className = resultOkWrapper.className.replace("hidden", "") + " hidden";
+      resultNotAllowed.className = resultNotAllowed.className.replace("hidden", "") + " hidden";
+      resultNetErr.className = resultNetErr.className.replace("hidden", "") + " hidden";
 
+      let submitButtonText = submitButton.innerText;
       let xhttp = new XMLHttpRequest();
-      let done = false;
       xhttp.onreadystatechange = function () {
         if (this.readyState === 4) {
-          done = true;
-          submitButton.innerText = submitButton.innerText.replace("...", "");
-          if (this.status === 200) {
-            // Success
-            window.localStorage.setItem("email", email);
-            form.className += " hidden";
-            resultOk.className = resultOk.className.replace("hidden", "");
-          } else {
-            // Failure
-            resultFail.className = resultFail.className.replace("hidden", "");
-          }
+          setTimeout(() => {
+            submitButton.innerText = submitButtonText
+            if (this.status === 200) {
+              // Success
+              setDatetimeElement(Date.now())
+              window.localStorage.setItem("email", email);
+              window.localStorage.setItem("checkin", `${btoa(email)}::${locationId}::${Date.now().toString()}`);
+              showCheckin(email)
+
+            } else if (this.status === 403) {
+              resultNotAllowed.className = resultNotAllowed.className.replace("hidden", "");
+
+            } else {
+              // Network or internal error
+              resultNetErr.className = resultNetErr.className.replace("hidden", "");
+            }
+          }, 700)
```

Comment:
Why always waiting?

Reply:
Ux / feels nicer than instantly getting the checkin screen.